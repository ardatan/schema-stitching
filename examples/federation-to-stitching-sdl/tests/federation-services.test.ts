import { gatewayApp } from '../src/gateway';
import { productsServer } from '../src/services/products/server';
import { reviewsServer } from '../src/services/reviews/server';
import { usersServer } from '../src/services/users/server';

describe('Federation services', () => {
  beforeAll(async () => {
    await Promise.all([
      new Promise<void>(resolve => productsServer.listen(4001, resolve)),
      new Promise<void>(resolve => reviewsServer.listen(4002, resolve)),
      new Promise<void>(resolve => usersServer.listen(4003, resolve)),
    ]);
  });
  afterAll(async () => {
    await Promise.all([
      new Promise(resolve => productsServer.close(resolve)),
      new Promise(resolve => reviewsServer.close(resolve)),
      new Promise(resolve => usersServer.close(resolve)),
    ]);
  });
  it('should work', async () => {
    const response = await gatewayApp.fetch('/graphql', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        query: /* GraphQL */ `
          query {
            user(id: "1") {
              username
              recentPurchases {
                upc
                name
              }
              reviews {
                body
                author {
                  id
                  username
                }
                product {
                  upc
                  name
                  acceptsNewReviews
                }
              }
            }
          }
        `,
      }),
    });
    const result = await response.json();
    expect(result).toMatchSnapshot('users');
  });
});
