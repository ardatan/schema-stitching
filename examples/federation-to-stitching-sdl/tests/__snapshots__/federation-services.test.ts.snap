// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Federation services should work: users 1`] = `
{
  "data": {
    "user": {
      "recentPurchases": [
        {
          "name": "iPhone",
          "upc": "1",
        },
        {
          "name": "iOS Survival Guide",
          "upc": "4",
        },
      ],
      "reviews": [
        {
          "author": {
            "id": "1",
            "username": "hanshotfirst",
          },
          "body": "Love it!",
          "product": {
            "acceptsNewReviews": true,
            "name": "iPhone",
            "upc": "1",
          },
        },
        {
          "author": {
            "id": "1",
            "username": "hanshotfirst",
          },
          "body": "Too expensive.",
          "product": {
            "acceptsNewReviews": false,
            "name": "Super Baking Cookbook",
            "upc": "2",
          },
        },
      ],
      "username": "hanshotfirst",
    },
  },
}
`;
